parties/opkclient.go

```diff
@@ -49,10 +78,15 @@ func (o *OpkClient) OidcAuth() {
 
 type TokenCallback func(tokens *oidc.Tokens[*oidc.IDTokenClaims])
 
+type PublicKey interface {
+	Equal(x crypto.PublicKey) bool
+}
+
 // Interface for interacting with the OP (OpenID Provider)
 type OpenIdProvider interface {
 	RequestTokens(cicHash string, cb TokenCallback) error
-	VerifyPKToken(pktCom []byte, cosPk *ecdsa.PublicKey) error
+	VerifyPKToken(pktJSON []byte, cosPk *ecdsa.PublicKey) error
+	PublicKey(idt []byte) (PublicKey, error)
```

Comment:
Oh, I really like this pattern you are using here. Adding the OP pubkey to the OpenIdProvider interface is great place to plug in different public key resolvers: JWKS URIs, JWT Ledger, TUFs, etc...

Reply:
Yep, it's a good way to have a test provider with a static key too